x-pack/solutions/security/plugins/security_solution/public/configurations/tabs/promotion_rules/promotion_rules_table.tsx

```diff
@@ -0,0 +1,225 @@
+    (ruleManagementFilters?.rules_summary.prebuilt_installed_count ?? 0);
+
+  const ruleTabs = useMemo(
+    () => [
+      {
+        id: PromotionRuleTabs.management,
+        name: i18n.INSTALLED_RULES_TAB,
+        append:
+          installedTotal > 0 ? (
+            <EuiNotificationBadge size="m" color="subdued">
+              {installedTotal}
+            </EuiNotificationBadge>
+          ) : undefined,
+        isSelected: currentTab === PromotionRuleTabs.management,
+        onClick: () => handleTabClick(PromotionRuleTabs.management),
+      },
+      {
+        id: PromotionRuleTabs.monitoring,
+        name: i18n.RULE_MONITORING_TAB,
+        append:
+          installedTotal > 0 ? (
+            <EuiNotificationBadge size="m" color="subdued">
+              {installedTotal}
+            </EuiNotificationBadge>
+          ) : undefined,
+        isSelected: currentTab === PromotionRuleTabs.monitoring,
+        onClick: () => handleTabClick(PromotionRuleTabs.monitoring),
+      },
+    ],
+    [currentTab, handleTabClick, installedTotal]
+  );
+
+  const shouldShowLinearProgress = (isFetched && isRefetching) || isUpgradingSecurityPackages;
+  const shouldShowLoadingOverlay = !isFetched && isRefetching;
+
+  return (
+    <>
+      <EuiTabs size="m" bottomBorder>
+        {ruleTabs.map((tab) => (
+          <EuiTab key={tab.id} {...tab}>
+            {tab.name}
+          </EuiTab>
+        ))}
+      </EuiTabs>
+      <EuiSpacer size="s" />
+      {shouldShowLinearProgress && (
+        <EuiProgress
+          data-test-subj="loadingRulesInfoProgress"
+          size="xs"
+          position="absolute"
+          color="accent"
+        />
+      )}
+      {shouldShowLoadingOverlay && (
+        <Loader data-test-subj="loadingPanelAllRulesTable" overlay size="xl" />
+      )}
+      <EuiBasicTable
+        itemId="id"
+        items={rules}
+        noItemsMessage={NO_ITEMS_MESSAGE}
+        onChange={tableOnChangeCallback}
+        pagination={paginationMemo}
+        sorting={{
+          sort: {
+            // EuiBasicTable has incorrect `sort.field` types which accept only `keyof Item` and reject fields in dot notation
+            field: sortingOptions.field as keyof Rule,
+            direction: sortingOptions.order,
+          },
+        }}
+        columns={rulesColumns}
+      />
+    </>
+  );
+};
+
+interface ColumnsProps {
+  currentTab: PromotionRuleTabs;
+}
+
+const useRulesColumns = ({ currentTab }: ColumnsProps): Array<EuiBasicTableColumn<Rule>> => {
+  const [{ canUserCRUD }] = useUserData();
+  const hasPermissions = hasUserCRUDPermission(canUserCRUD);
+
+  const enabledColumn = useEnabledColumn({
+    hasCRUDPermissions: hasPermissions,
+    isLoadingJobs: false,
+    mlJobs: [],
+    startMlJobs: async (jobIds: string[] | undefined) => {},
+  });
+  const executionStatusColumn = useRuleExecutionStatusColumn({
+    sortable: true,
+    width: '16%',
```

Comment:
Fixed width should work better. Otherwise min width should be set. Relative width gets too narrow on smaller screens

<img width="946" alt="image" src="https://github.com/user-attachments/assets/431d41ba-e485-459a-b581-12e2eef06941" />